File: HomeWork_3/src/com/valery/homework3/PartOne/Publication.java
package com.valery.homework3.PartOne;

public interface Publication {
    double getPrice();
    String getType();
    int getWordsCount();

    default void description() {
        System.out.println();
        System.out.printf("This publication's type is %s\n", getType());
        System.out.printf("This publication has %d words\n", getWordsCount());
        System.out.printf("This publication costs €%5.2f\n", getPrice());
    }
}

File: HomeWork_1/src/main/kotlin/Publication.kt
interface Publication {
    val price: Double
    val wordCount: Int
    fun getType(): String
}

fun Publication.description() {
    println("This publication's type is ${getType()}")
    println("This publication has $wordCount words")
    println("This publication costs €$price\n")
}

fun buy(publication: Publication) {
    println("The purchase is complete. The purchase amount was €${publication.price}\n")
}